---
layout: archive
title: Zulip Chat Archive
permalink: /stream/113488-general/topic/Weird.20CI.20failure.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/113488-general/index.html">general</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html">Weird CI failure</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="246596607"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246596607" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246596607">(Jul 20 2021 at 14:35)</a>:</h4>
<p>I'm getting <a href="https://github.com/leanprover-community/mathlib/pull/8375/checks?check_run_id=3114964848">the following CI failure</a> that I don't understand:</p>
<div class="codehilite" data-code-language="Text only"><pre><span></span><code>Setting up auth
  /usr/bin/git config --local --name-only --get-regexp core\.sshCommand
  /usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' &amp;&amp; git config --local --unset-all 'core.sshCommand' || :
  Error: fatal: No url found for submodule path 'class-number' in .gitmodules
  Error: The process '/usr/bin/git' failed with exit code 128
</code></pre></div>
<p>The only relation I can think of is that the Lean Forward organisation has a separate <code>class-number</code> repository that has mathlib as a dependency, but this commit was made on a clean copy of mathlib master. And neither repository has a <code>.gitmodules</code> file AFAICT</p>



<a name="246597416"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246597416" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246597416">(Jul 20 2021 at 14:37)</a>:</h4>
<p>Other CI jobs with the same "Setting up auth" rule pass, but restarting the build doesn't seem to help.</p>



<a name="246598612"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246598612" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Oliver Nash <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246598612">(Jul 20 2021 at 14:42)</a>:</h4>
<p>Likewise I've suddenly started hitting this and I'm very confused. It is surely a coincidence but I've been tidying up <a href="https://github.com/leanprover-community/mathlib/tree/p-adic">branch#p-adic</a> and just removed an empty folder (in that branch) called <code>class-number</code>!</p>



<a name="246598870"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246598870" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246598870">(Jul 20 2021 at 14:42)</a>:</h4>
<p><a href="https://github.com/leanprover-community/mathlib/runs/3115085697#step:2:149">https://github.com/leanprover-community/mathlib/runs/3115085697#step:2:149</a> is the same</p>



<a name="246599159"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246599159" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Johan Commelin <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246599159">(Jul 20 2021 at 14:43)</a>:</h4>
<p>Is there any reason why <code>class-number</code> could have ended up on the CI runners?<br>
And is there a chance that CI runs aren't cleaning up after themselves well enough?</p>



<a name="246599896"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246599896" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246599896">(Jul 20 2021 at 14:45)</a>:</h4>
<p>I haven't touched the <code>class-number</code> repo in a couple of weeks, it's in a different organisation and I don't think it involves any submodules. So I don't think that's the cause.</p>



<a name="246599959"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246599959" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246599959">(Jul 20 2021 at 14:46)</a>:</h4>
<p>CI runners keeping around a bit too much state sounds possible.</p>



<a name="246600369"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246600369" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246600369">(Jul 20 2021 at 14:47)</a>:</h4>
<p>Git doesn't track directories, so removing an empty directory from the repository sounds a bit suspicious...</p>



<a name="246600559"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246600559" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Oliver Nash <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246600559">(Jul 20 2021 at 14:48)</a>:</h4>
<p>Here's the commit where I remove an empty folder called <code>class-number</code>: <a href="https://github.com/leanprover-community/mathlib/commit/de03792d96f51fccc2813f98c156dca62b803173">https://github.com/leanprover-community/mathlib/commit/de03792d96f51fccc2813f98c156dca62b803173</a> but I _now_ note that earlier commits on that branch also hit the same error.</p>



<a name="246601168"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246601168" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246601168">(Jul 20 2021 at 14:50)</a>:</h4>
<p>Yeah, looking at the diff saying "Submodule class-number deleted from 6330d4" this seems to be almost directly connected.</p>



<a name="246601845"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246601845" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246601845">(Jul 20 2021 at 14:55)</a>:</h4>
<p><a href="https://github.com/actions/checkout/issues/354">https://github.com/actions/checkout/issues/354</a></p>



<a name="246601923"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246601923" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246601923">(Jul 20 2021 at 14:55)</a>:</h4>
<p>And also <a href="https://github.com/actions/checkout/issues/385">https://github.com/actions/checkout/issues/385</a></p>



<a name="246601932"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246601932" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246601932">(Jul 20 2021 at 14:55)</a>:</h4>
<p>Which is specific to self-hoster runners</p>



<a name="246602259"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246602259" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Johan Commelin <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246602259">(Jul 20 2021 at 14:58)</a>:</h4>
<p>Good find! That sounds very related to the problems we are seeing!</p>



<a name="246602335"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246602335" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246602335">(Jul 20 2021 at 14:58)</a>:</h4>
<p>So for now we can just clean up the worker and hope no one ever makes a submodule again</p>



<a name="246602346"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246602346" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Anne Baanen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246602346">(Jul 20 2021 at 14:58)</a>:</h4>
<p>Issue 385 seems to be exactly what is going on here. Would a CI step that removes <code>.gitmodules</code> before checkout fix this?</p>



<a name="246604251"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246604251" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Bryan Gin-ge Chen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246604251">(Jul 20 2021 at 15:09)</a>:</h4>
<p>Seems worth trying!</p>



<a name="246606634"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Weird%20CI%20failure/near/246606634" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Bryan Gin-ge Chen <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Weird.20CI.20failure.html#246606634">(Jul 20 2021 at 15:26)</a>:</h4>
<p>(I should add that I can't work on this until tonight, so if someone wants to take a shot at it, please do!)</p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>